libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -109,46 +109,26 @@ std::pair<bool, std::string> TransactionManager::insertTransaction(const std::sh
   }
 }
 
-uint32_t TransactionManager::insertValidatedTransactions(
-    std::vector<std::pair<std::shared_ptr<Transaction>, TransactionStatus>> &&txs) {
-  std::vector<std::pair<std::shared_ptr<Transaction>, TransactionStatus>> unseen_txs;
-  std::vector<trx_hash_t> txs_hashes;
-
-  if (txs.empty()) {
-    return 0;
-  }
-
-  txs_hashes.reserve(txs.size());
-  std::transform(txs.begin(), txs.end(), std::back_inserter(txs_hashes),
-                 [](const auto &t) { return t.first->getHash(); });
+bool TransactionManager::insertValidatedTransaction(std::pair<std::shared_ptr<Transaction>, TransactionStatus> &&tx) {
+  const auto trx_hash = tx.first->getHash();
 
   // This lock synchronizes inserting and removing transactions from transactions memory pool with database insertion.
   // It is very important to lock checking the db state of transaction together with transaction pool checking to be
   // protected from new DAG block and Period data transactions insertions which are inserted directly in the database.
   std::unique_lock transactions_lock(transactions_mutex_);
   const auto last_block_number = final_chain_->last_block_number();
 
-  // Check the db with a multiquery if transactions are really new
-  auto db_seen = db_->transactionsInDb(txs_hashes);
-  for (uint32_t i = 0; i < txs_hashes.size(); i++) {
-    if (!db_seen[i]) {
-      unseen_txs.push_back(std::move(txs[i]));
-    } else {
-      // In case we received a new tx that is already in db, mark it as known in cache
-      transactions_pool_.markTransactionKnown(txs[i].first->getHash());
-    }
+  // Check the db with if transaction is really new
+  if (db_->transactionInDb(trx_hash)) {
+    transactions_pool_.markTransactionKnown(trx_hash);
+    return false;
   }
 
-  size_t trx_inserted_count = 0;
-  // Save transactions in memory pool
-  for (auto &trx : unseen_txs) {
-    auto tx_hash = trx.first->getHash();
-    LOG(log_dg_) << "Transaction " << tx_hash << " inserted in trx pool";
-    if (transactions_pool_.insert(std::move(trx), last_block_number)) {
-      trx_inserted_count++;
-    }
+  LOG(log_dg_) << "Transaction " << trx_hash << " inserted in trx pool";
+  if (transactions_pool_.insert(std::move(tx), last_block_number)) {
```

Comment:
you could use insert as return ? :
return transactions_pool_.insert(std::move(tx), last_block_number)

Reply:
Fixed